codalab/worker/worker.py

```diff
@@ -474,8 +474,11 @@ def process_runs(self):
 
         # 1. transition all runs
         for uuid in self.runs:
-            run_state = self.runs[uuid]
-            self.runs[uuid] = self.run_state_manager.transition(run_state)
+            prev_state = self.runs[uuid]
+            self.runs[uuid] = self.run_state_manager.transition(prev_state)
+            if prev_state.stage != self.runs[uuid].stage:
```

Comment:
Add a comment here explaining what this if statement does, and why we want to run this code only in this if statement

Reply:
done